src/core/lib/channel/channel_args.h

```diff
@@ -63,14 +68,71 @@ struct ChannelArgTypeTraits<
         [](void* p) { static_cast<T*>(p)->Unref(); },
         // compare
         [](void* p1, void* p2) {
-          return QsortCompare(*static_cast<const T*>(p1),
-                              *static_cast<const T*>(p2));
+          return T::ChannelArgsCompare(static_cast<const T*>(p1),
+                                       static_cast<const T*>(p2));
         },
     };
     return &tbl;
   };
 };
 
+template <typename T>
+struct ChannelArgTypeTraits<
+    T, absl::enable_if_t<
+           std::is_base_of<RefCounted<T, NonPolymorphicRefCount>, T>::value,
+           void>> {
+  static const grpc_arg_pointer_vtable* VTable() {
+    static const grpc_arg_pointer_vtable tbl = {
+        // copy
+        [](void* p) -> void* { return static_cast<T*>(p)->Ref().release(); },
+        // destroy
+        [](void* p) { static_cast<T*>(p)->Unref(); },
+        // compare
+        [](void* p1, void* p2) {
+          return T::ChannelArgsCompare(static_cast<const T*>(p1),
+                                       static_cast<const T*>(p2));
+        },
+    };
+    return &tbl;
+  };
+};
+
+template <typename T>
+struct ChannelArgTypeTraits<
```

Comment:
This specialization appears to be same (in terms of the implementation of copy, destroy and compare functions) as the previous two. Is it to allow future expansions ? or can they be combined ?